infrastructure/lib/constructs/lambda-fn.ts

```diff
@@ -0,0 +1,219 @@
+import * as ec2 from "@aws-cdk/aws-ec2";
+import * as iam from "@aws-cdk/aws-iam";
+import * as secretsmanager from "@aws-cdk/aws-secretsmanager";
+import * as lambda from "@aws-cdk/aws-lambda";
+import * as lambdaNodeJs from "@aws-cdk/aws-lambda-nodejs";
+import * as cdk from "@aws-cdk/core";
+import * as s3 from "@aws-cdk/aws-s3";
+import { HttpMethod } from "../http";
+import * as dynamodb from "@aws-cdk/aws-dynamodb";
+import { SecureBucket } from "./compliant-resources";
+import * as sqs from "@aws-cdk/aws-sqs";
+import { SqsEventSource } from "@aws-cdk/aws-lambda-event-sources";
+import * as sfn from "@aws-cdk/aws-stepfunctions";
+
+/**
+ * An IAM service principal for the API Gateway service, used to grant Lambda
+ * invocation permissions.
+ */
+const APIGW_SERVICE_PRINCIPAL = new iam.ServicePrincipal("apigateway.amazonaws.com");
+
+export interface ApiFunctionPropstest {
```

Comment:
Why is `test` at the end of this name? Why do the `Props` not match the name of the class as we do everywhere else?

Reply:
I think `ApiFunction` was an obvious choice but it's used in infrastructure, and I'm guessing that this was a lazy way to create a unique name.  Zach was looking for a better name for the construct.  I made a few suggestions elsewhere.

Re: props name, I suppose that the props for `ApiFunctiontest` should be `ApiFunctiontestProps` if keeping with the convention.